Resources/ChunkData.gd

```diff
@@ -12,26 +15,38 @@ func _ready() -> void:
 
 func generate_height_map() -> void:
 	height_map.resize(width * width)
-	var max_height: float = 0
-	var min_height: float = INF
 
 	for x in range(width):
 		for z in range(width):
-			var index: int = x + z * width
+			var index: int = z + x * width
 			var height: float = noise.get_noise_2d(x, z)
 
-			if height > max_height:
-				max_height = height
-
-			if height < min_height:
-				min_height = height
-			
+			height = (height + 1.0) / 2.0
 			height_map[index] = height
-	_normalize_height_map(min_height, max_height)
+	_draw_color_map()
+	_height_enhancement()
 
 
-func _normalize_height_map(min_height: float, max_height: float) -> void:
+func _height_enhancement() -> void:
 	for x in range(width):
 		for z in range(width):
-			var _index: int = x + z * width
-			height_map[_index] = inverse_lerp(min_height, max_height, height_map[_index])
+			var index: int = z + x * width
+			height_map[index] = height_map[index] * 10
+
+
+func _draw_color_map() -> void:
+	var image := Image.create_empty(width, width, false, Image.FORMAT_RGB8)
+	for x in range(width):
+		for z in range(width ):
+			var index: int = z + x * width
+			var height: float = height_map[index]
+			for key: Color in regions.keys():
+				if height <= regions[key]:
```

Comment:
You may want to consider using a gradient for your regions for a smoother image.
Unless what you are going for is that blocky hash cutoff look.

Better yet, you could make it a part of a shader!
But if you plan on moving towards more complex texture generation, then you can ignore this comment.

Reply:
That could be really nice actually, the gradient, maybe the shader too. 
Though eventually I'd like to have gradients between different textures, but not colors.